validator/lib/utility.js

```diff
@@ -58,14 +62,53 @@ function regionInsideRegion(regionInner, regionOuter) {
         (regionInner[5] <= regionOuter[5]);
 }
 
-var scratchInnnerCenter = new Cartesian3();
+var scratchInnerCenter = new Cartesian3();
 var scratchOuterCenter = new Cartesian3();
 
 function sphereInsideSphere(sphereInner, sphereOuter) {
     var radiusInner= sphereInner[3];
     var radiusOuter = sphereOuter[3];
-    var centerInner = Cartesian3.unpack(sphereInner, 0, scratchInnnerCenter);
+    var centerInner = Cartesian3.unpack(sphereInner, 0, scratchInnerCenter);
     var centerOuter = Cartesian3.unpack(sphereOuter, 0, scratchOuterCenter);
     var distance = Cartesian3.distance(centerInner, centerOuter);
     return distance <= (radiusOuter - radiusInner);
 }
+
+var scratchInnerHalfAxes = new Matrix3();
+var scratchOuterHalfAxes = new Matrix3();
+
+function boxInsideBox(boxInner, boxOuter) {
+    var centerInner = Cartesian3.fromElements(boxInner[0], boxInner[1], boxInner[2], scratchInnerCenter);
+    var halfAxesInner = Matrix3.fromArray(boxInner, 3, scratchInnerHalfAxes);
+    var transformInner = Matrix4.fromRotationTranslation(halfAxesInner, centerInner);
+
+    var centerOuter = Cartesian3.fromElements(boxOuter[0], boxOuter[1], boxOuter[2], scratchOuterCenter);
+    var halfAxesOuter = Matrix3.fromArray(boxOuter, 3, scratchOuterHalfAxes);
+    var transformOuter = Matrix4.fromRotationTranslation(halfAxesOuter, centerOuter);
+
+    var cube = new Array(8);
+    cube[0] = new Cartesian3(-1, -1, -1);
+    cube[1] = new Cartesian3(-1, -1, 1);
+    cube[2] = new Cartesian3(1, -1, 1);
+    cube[3] = new Cartesian3(1, -1, -1);
+    cube[4] = new Cartesian3(-1, 1, -1);
+    cube[5] = new Cartesian3(-1, 1, 1);
+    cube[6] = new Cartesian3(1, 1, 1);
+    cube[7] = new Cartesian3(1, 1, -1);
+
+    var i = 0;
+    var transformInnerInverse;
+    var min, max;
+    var EPSILON8 = CesiumMath.EPSILON8;
+    for (i = 0; i < 8; i++) {
```

Comment:
Now that there is one for loop, it is fine to do `for (var i = 0`.

Similarly `min` and `max` do not need to be declared outside of the loop.